courses-app/src/app/shared/components/header/header.component.html

```diff
@@ -4,24 +4,29 @@
       <img alt="Courses Logo" src="../../../../assets/bootstrap.png">
       <a [routerLink]="['/courses']" class="navbar-logo-text">Courses</a>
     </div>
-    <ul *ngIf="isAuthorized; else guest" class="navbar-menu">
-      <li class="navbar-menu-element">
-        <span>{{userName}} - {{userRole}}</span>
-      </li>
-      <li class="navbar-menu-element">
-        <app-button (click)="logout()"
-                    [buttonIcon]="'fa fa-sign-out'"
-                    [buttonText]="'Logout'"></app-button>
-      </li>
-    </ul>
-    <ng-template #guest>
-      <ul class="navbar-menu">
+    <div *ngIf="((userStateFacade.$isLoading | ngrxPush) || (authStateFacade.$isLoading | ngrxPush)); else headerData">
+      <app-spinner></app-spinner>
+    </div>
+    <ng-template #headerData>
+      <ul *ngIf="authStateFacade.$isAuthorized | ngrxPush; else guest" class="navbar-menu">
+        <li class="navbar-menu-element">
+          <span>{{userStateFacade.$name | ngrxPush}} - {{userStateFacade.$role | ngrxPush}}</span>
+        </li>
         <li class="navbar-menu-element">
-          <app-button [buttonIcon]="'fa fa-sign-out'"
-                      [buttonText]="'Login'"
-                      [routerLink]="['/login']"></app-button>
+          <app-button (click)="logout()"
+                      [buttonIcon]="'fa fa-sign-out'"
```

Comment:
If bindings are static ,

`buttonIcon="fa fa-sign-out"`

is enough

Pro hint: read about https://indepth.dev/posts/1315/angular-cdk-coercion and https://material.angular.io/cdk/coercion/overview , very useful when implementing inputs